logprep/connector/opensearch/output.py

```diff
@@ -136,7 +143,24 @@ def _bulk(self, client, actions, *args, **kwargs):
         except search.exceptions.TransportError as error:
             self._handle_transport_error(error)
 
-    def _parallel_bulk(self, client, actions, *args, **kwargs):
+    def _parallel_bulk_with_retries(self, client, actions, *args, **kwargs):
+        bulk_delays = 1
+        for _ in range(self._config.bulk_retries):
+            try:
+                self._parallel_bulk(actions, client, *args, **kwargs)
+                return
+            except search.ConnectionError as error:
+                raise error
+            except search.exceptions.TransportError as error:
+                if self._message_exceeds_max_size_error(error):
+                    raise error
+                time.sleep(bulk_delays)
+                bulk_delays += random.randint(500, 1000) / 1000
+        raise FatalOutputError(
+            self, "Opensearch too many requests, all parallel bulk retries failed"
+        )
```

Comment:
instead of writing this new method, how about extending the called method and add the `max_retries` parameter to the signature and use it?

Reply:
well I implemented it as a second method to reduce nesting. I reverted it now, please have a look at it and let me know which version you like more: (a) all in one method or (b) split up into two methods.